lib/MusicBrainz/Server/Entity/Relationship.pm

```diff
@@ -279,9 +281,9 @@ around TO_JSON => sub {
         entity1_credit  => $self->entity1_credit,
         entity0_id      => $self->entity0_id,
         entity1_id      => $self->entity1_id,
-        id              => $self->id + 0,
-        linkOrder       => $self->link_order + 0,
-        linkTypeID      => $link->type_id + 0,
+        id              => $self->id ? $self->id + 0 : 0,
+        linkOrder       => $self->link_order ? $self->link_order + 0 : 0,
+        linkTypeID      => $link->type_id ? $link->type_id + 0 : 0,
```

Comment:
For `id`/`linkTypeID` I think it makes more sense to output `undef` (null) if they're unset, not 0, so that we don't have a special numeric case there. For `linkOrder` outputting 0 is fine since that's the default value at the database level.